src/Container.php

```diff
@@ -67,12 +70,21 @@ public function __construct(
 	 * @template ID
 	 * @param class-string<ID> $id The ID of the class to retrieve.
 	 * @return ID
+	 * @throws ContainerException In case the ID could not be resolved or instantiated.
 	 *
 	 * Psalm expects $id to be a string, based on ContainerInterface.
 	 * @psalm-suppress MoreSpecificImplementedParamType
+	 *
+	 * PSR-11 containers declares to throw an un-throwable interface
+	 * (it does not extend Throwable), and Psalm does not accept it.
+	 * @psalm-suppress MissingThrowsDocblock
 	 */
 	public function get( $id ) {
-		return $this->container->get( $id );
+		try {
+			return $this->container->get( $id );
+		} catch ( Throwable $e ) {
+			throw new ContainerException( $e->getMessage() );
+		}
```

Comment:
This was supposed to be a small side fix in the original PR. That's because currently we don't have much capacity to quickly iterate over PRs, but we still need to maintain project velocity.

> I remember you mentioning that the container doesn't throw Throwables, and yet here you catch one and wrap it in a custom exception

It does indeed always throw throwables, but that's not how it is declared. That's why I'm catching what's partially undefined, and throwing another, properly traceable exception.

Anyway, that's the topic for a different discussion.